lib/errors.js

```diff
@@ -0,0 +1,13 @@
+'use strict'
+
+class FastifyGraphQLError extends Error {
+  constructor (message, code, additionalProperties) {
+    super(message)
+    this.code = code
+    this.additionalProperties = additionalProperties
+  }
+}
+
+module.exports = {
+  FastifyGraphQLError: FastifyGraphQLError
+}
```

Comment:
I would call this `ErrorWithProps` or similar and not have the name of the framework in it.